### PR TITLE
Nested if then else

### PR DESCRIPTION
**Pull request by allevo**
Implement nested if then else as the test explains

https://github.com/fastify/fast-json-stringify/blob/0f3e9726c141de6dea10feba9717b51967973bf9/test/if-then-else.test.js#L31-L67

NB: the developer shouldn't mention the property twice

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -512,57 +514,82 @@ function buildInnerObject (schema, name, externalSchema, fullSchema) {
...
+  var r
+  var laterCode = ''
+  var innerR
+
+  const i = schema.if
+  const then = schema.then
+  const e = schema.else
+  delete schema.if
+  delete schema.then
+  delete schema.else
```
Comment (mcollina) [inline index.js]: we might want to clone this before deleting.

Comment (allevo) [reply index.js]: why?

Comment (allevo) [reply index.js]: Good point I clone the schema deeply

Comment (allevo) [reply index.js]: done

**Pull request merged**